### DISCUSSION TITLE
Next.js using `app.UseSpa` for ASP.NET Core?

### DISCUSSION THREAD

**Discussion by NateRadebaugh** [vercel/next.js#21571] [Help]
My company uses ASP.NET for a set of client projects. We're able to use angular using the Microsoft SPA templates, but I'd like to switch over to Next.js for the front-end.

I've got a solution that works in development using `spa.UseProxyToSpaDevelopmentServer` but I don't think this is the right thing to use in production.

Options:
1. `next export` and serve static files using `app.UseSpaStaticFiles();` in `Startup.cs`
   * Issue: `next/image` doesn't support `next export`, so this doesn't work.
2. `next build` and figure out proper `Startup.cs` config to work correctly in `production`.
3. ??

Has anyone had success in this setup? As you'd expect, I'm hosting on Azure for these C# solutions.

**Comment by spong-e** [comment]
@NateRadebaugh hi, I’m literally having the exact same issue. Did you find a solution? Thanks

**Reply by NateRadebaugh** [reply]
No

**Reply by Ohmniox** [reply]
@NateRadebaugh @spong-e Did you guys found out anything? I am having the same issue :(

**Reply by spong-e** [reply]
Hi @Ohmniox, sadly no, actually ended up not using Next and just using React. The reason being, we werent able to use node to run the next app, so routing, etc wasnt going to give us what we needed.

**Reply by TDiblik** [reply]
Hi, for anybody finding this, user @davidnx has made a promising [nuget package](https://www.nuget.org/packages/NextjsStaticHosting.AspNetCore), along with [blog post](https://medium.com/@david.nissimoff/next-js-meets-asp-net-core-a-story-of-performance-and-love-at-long-tail-41cf9231b2de). You can find source code [here](https://medium.com/@david.nissimoff/next-js-meets-asp-net-core-a-story-of-performance-and-love-at-long-tail-41cf9231b2de). I looked into the source a bit and it seems like it could work. I have yet to try it tho (will in like a week).
